### PR TITLE
build: migrate to MDX v2

### PR DESCRIPTION
**Pull request by axe312ger**
This upgrades the website to MDX v2. We use this repo to test our canary progress :)

See: https://github.com/gatsbyjs/gatsby/pull/35650

### CODE REVIEW

File: gatsby-config.js [hunk 1]
```diff
@@ -106,28 +109,29 @@ module.exports = {
...
-            options: {
-              wrapperStyle: `margin-bottom: 1.0725rem`,
-            },
-          },
-          // 'gatsby-remark-copy-linked-files',
-          'gatsby-remark-smartypants',
-        ],
-      },
-    },
+    // Are these maybe leftovers from pre-MDX times? Or have these plugins been exectued on MDX (v1) nodes as well?
```
Comment (axe312ger) [inline gatsby-config.js:112]: This is an open question. I think all of this can be removed?

**Comment by wesbos** [comment]
Thanks - just approved it, will see if it builds!

**Comment by axe312ger** [comment]
@wesbos we are now in release candidate status.

It builds on Gatsby Cloud, could you please check if your Netlify wants to build? <3

**Comment by axe312ger** [comment]
@wesbos could you enable Netlify builds for this PR now pls? :)

**Comment by wesbos** [comment]
@axe312ger is this good to merge now?

**Comment by axe312ger** [comment]
@wesbos kind of depends if this is relevant for you:

https://github.com/wesbos/wesbos/pull/325/files#diff-b5e305780d9d473da97c61beab8bc36e5e8871b360942e4686c9b20d8c5d4cfaL81-R95

**Comment by wesbos** [comment]
I did end up using parts of this PR to upgrade - thank you

**Comment by wesbos** [comment]
Seems I thought I was done the V2 migration, but I'm not :( re-opening

**Comment by wesbos** [comment]
WHEW! I got it! https://github.com/wesbos/wesbos/pull/371

Thanks for the PR on this - really helped solve a lot of my issues.

**Pull request closed without merging**